iothub/device/tests/DeviceClientTwinApiTests.cs

```diff
@@ -59,6 +59,27 @@ await innerHandler.
             Assert.AreEqual(client.InternalClient._desiredPropertyUpdateCallback, myCallback);
         }
 
+        [TestMethod]
+        public async Task DeviceClientDesiredPropertyUpdateCallbackUnsubscribes()
+        {
+            // arrange
+            var innerHandler = Substitute.For<IDelegatingHandler>();
+            var client = DeviceClient.CreateFromConnectionString(fakeConnectionString);
+            client.InnerHandler = innerHandler;
+            DesiredPropertyUpdateCallback myCallback = (p, c) => TaskHelpers.CompletedTask;
+            var context = new object();
+
+            // act
+            await client.SetDesiredPropertyUpdateCallbackAsync(myCallback, context).ConfigureAwait(false);
```

Comment:
same here, this call can also be skipped.

Reply:
No we need to set before unset.